### PR TITLE
Remove `const` ttype

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lfortran/lfortran/issues/3374

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
This is a draft PR and the tests might fail.

**Comment by certik** [comment]
This must be done especially in LPython. We need to sync the ASR as well.

**Comment by Shaikh-Ubaid** [comment]
Yes.

File: integration_tests/template_array_03.f90 [hunk 1]
```diff
@@ -84,7 +84,7 @@ subroutine mymatmul_t(i, j, k, a, b, r)
             integer, parameter, intent(in) :: i, j, k
             type(t), intent(in) :: a(i,j), b(j,k)
             type(t) :: r(i,k)
-            integer, parameter :: x = 1, y = 1, z = 1
+            integer :: x = 1, y = 1, z = 1
```
Comment (Shaikh-Ubaid) [inline integration_tests/template_array_03.f90:87]: These variables are used as loop variables and thus need to get updated. Thus, made them non-const.

**Review by czgdp1807 — APPROVED** [review]

File: integration_tests/template_array_03.f90 (hunk 1 above)
Comment (czgdp1807) [reply integration_tests/template_array_03.f90:87]: How were they working with GFortran till now? This is weird.

Comment (Shaikh-Ubaid) [reply integration_tests/template_array_03.f90:87]: It is a template test case. It seems we do not run template tests with `gfortran`.

**Comment by Shaikh-Ubaid** [comment]
Thanks for the review.

File: integration_tests/template_array_03.f90 (hunk 1 above)
Comment (Shaikh-Ubaid) [reply integration_tests/template_array_03.f90:87]: Related https://github.com/lfortran/lfortran/issues/3739#issuecomment-2027402370.

**Comment by certik** [comment]
@Shaikh-Ubaid can you please do the same change in LPython?

**Pull request merged**